syncd/Makefile.am

```diff
@@ -37,6 +37,10 @@ syncd_CPPFLAGS += -DSAITHRIFT=yes
 syncd_LDADD += -lrpcserver -lthrift
 endif
 
+if sonic_asic_platform_mellanox
+syncd_CPPFLAGS += -DUNINIT_DATA_PLANE_ON_REMOVAL
```

Comment:
>UNINIT_DATA_PLANE_ON_REMOVAL [](start = 20, length = 28)

suggest macro name: SAI_SUPPORT_UNINIT_DATA_PLANE_ON_REMOVAL
to indicate this is a feature, not a specific action.

The UNINIT_DATA_PLANE_ON_REMOVAL may indicate that you want to destroy data plane on removal for Mellanox ASIC.